.github/workflows/cd.yml

```diff
@@ -0,0 +1,78 @@
+name: Continuous Deployment
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pre-commit:
+    name: ruff and hooks.
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
+      - uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: --all-files --show-diff-on-failure
+
+  cd:
+    name: CD
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        lfs: false
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.8'
+
+    - name: Cache dependency
+      id: cache-dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[dev]
+
+    - name: Check types with pyright
+      run: |
+        pyright sbi
```

Comment:
In case we test on multiple Python versions, we should run pyright only on our "dev Python version". See #1102

Reply:
I never used pyright before, but this seems like sth we should do in the CI workflow where there is a matrix test. As in my comment above, I am confident that it is best to limit the CD workflow to the Python version that is best suited/supported.
Which change should be done to the command in line 59 then?

Reply:
Another (minor) reason to not do this in CD. How about the CI workflow? I think I copied this line over from the `tests.yml`. Should we change sth there?

Reply:
Right now, we don't have to change anything here. It's just important that we only run pyright on the Python version we define as our "dev version".